Have sync15_passwords_sync and callers return the sync ping.

mhammond opened the pull request:
This is towards #1154, and I *think* has most of the Android parts working - I'll make a simple android-components PR too. There's some iOS changes and I guessed the syntax, so probably doesn't even compile, but I'll test asap :) Probably wants a changelog entry too?

thomcc commented:
This should fix your Swift code.

```patch
diff --git a/components/logins/ios/Logins/LoginsStorage.swift b/components/logins/ios/Logins/LoginsStorage.swift
index d9f73904..50f6c376 100644
--- a/components/logins/ios/Logins/LoginsStorage.swift
+++ b/components/logins/ios/Logins/LoginsStorage.swift
@@ -141,7 +141,7 @@ open class LoginsStorage {
     open func sync(unlockInfo: SyncUnlockInfo) throws -> String {
         return try queue.sync {
             let engine = try self.getUnlocked()
-            return try LoginsStoreError.unwrap { err in
+            let ptr = try LoginsStoreError.unwrap { err in
                 sync15_passwords_sync(engine,
                                       unlockInfo.kid,
                                       unlockInfo.fxaAccessToken,
@@ -149,6 +149,7 @@ open class LoginsStorage {
                                       unlockInfo.tokenserverURL,
                                       err)
             }
+            return String(freeingRustString: ptr)
         }
     }
 ```

mhammond commented:
I updated with the review comments, and also updated https://github.com/mozilla-mobile/android-components/pull/3219 - but I can't seem to make lockwise understand the SyncTelemetryPing type.

mhammond commented:
I'm making some progress here - I've got the a-c logins sample logging the FxA UID, so I think this is ready to merge, and once merged and released, it will make the a-c PR much simpler.

thomcc commented on components/logins/android/build.gradle:
```diff
@@ -88,6 +88,9 @@ configurations {
 
 dependencies {
+    // Part of the public API.
+    api project(':sync15-library')
```
Hmm... Need to think if the issues the current sync15-library gradle set up (ugh) has are going to be a problem here. Since you've tested this in lockwise I'm guessing something is making it work regardless, so it's... probably fine?

Even if they are, an issue we could always backport the fixes from the megazord-only patch to avoid making this patch depend on that. Might want to do that anyway before adding more stuff that depends on sync15-library.

That or wait for the megazord-only patch to land, although it hasn't gotten review yet so.

thomcc commented:
> I'm making some progress here - I've got the a-c logins sample logging the FxA UID, so I think this is ready to merge, and once merged and released, it will make the a-c PR much simpler.

Ah, I misread this and thought it was tested in lockwise. I suspect it wouldn't work, and you'd have the issues @linacambridge had when integrating telemetry into fenix. In that case, it might be better to wait on the megazord-only patch actually (since I don't think the backport mentioned above will be enough without also fixing some things in lockwise's build config)

Alternatively, we could just all stop using gradle, because it's bad.

mhammond commented:
Thanks for your help Thom. I've *nearly* got lockwise working, but now it's failing at runtime with:

`java.lang.RuntimeException: Unable to create application mozilla.lockbox.LockboxApplication: java.lang.RuntimeException: LibViaduct not available`

which I'll dig into, but it seems like that's not at all related to this patch - so I think this is probably OK to land?
